### ISSUE TITLE
`read_bson` errors with relative paths

### ISSUE THREAD

**Issue by vrongmeal** [GlareDB/glaredb#2408]
## Description

```
> select * from read_bson('abc.bson');
Error: External error: Object at location /abc.bson not found: No such file or directory (os error 2)
```

**Comment by scsmithr** [comment]
To add, `./` also fails

```
> select * from read_bson('./hello.bson');
Error: External error: Error parsing Path "./hello.bson": Encountered illegal character sequence "." whilst parsing path segment "."
```

**Comment by hemanth94** [comment]
Hello, @tychoish 
Is this #2408 fixed? I see a PR closed related to this #2594 
But I'm still having the below error when i run `read_bson` locally, with latest v0.8.4.

In my windows.
![Screenshot 2024-02-12 155845](https://github.com/GlareDB/glaredb/assets/17337455/135695fa-f334-4b23-9a26-66db4330395a)

I want to try fixing this issue if it is unresolved. 
Thanks,

**Comment by tychoish** [comment]
Interesting!

I think the cases you've tried here are not the same as the ones I tested for, which were the ones without the `./` prefix. 

I think, having said that, that the current bson handling code in `main` is the same code we're using in other data sources, so this is surprising to me. 

I'm also aware that we're not testing on Windows, so that might be an issue. 

There's this related issue #2520 that I think might impact `read_bson` as well.
